### ISSUE TITLE
Install DEC APP run command "npm i --production --force" failed

### ISSUE THREAD

**Issue by lizhihongTest** [buckyos/CYFS#216]
**Describe the bug**
Install DEC APP run command "npm i --production --force" failed
"npm i --production --force" just install package.json::dependencies and ignore devDependencies , This command is correct for npm but seems incompatible in CYFS app-manager

```
[2023-04-15 09:21:10.298838 +00:00] INFO [ThreadId(8)] [service/app-manager/src/docker_api.rs:80] will spawn cmd: docker run --name decapp-9tgplnniojkx6rvcddi7t6bk1smqdjw32rfhxqureifh-install --cap-add NET_ADMIN --cap-add NET_RAW --rm --init --network bridge -v /cyfs/log/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/cyfs/log/app -v /cyfs/data/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/cyfs/data/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH -v /cyfs/app/dockerfile/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH/start.sh:/opt/start.sh:ro -v /cyfs/tmp:/cyfs/tmp -v /etc/localtime:/etc/localtime:ro -v /run/systemd/resolve/resolv.conf:/etc/resolv.conf:ro -v /cyfs/app/9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH:/opt/app buckyos/dec-app-base:v1 npm i --production --force
[2023-04-15 09:21:10.523285 +00:00] ERROR [ThreadId(8)] [service/app-manager/src/docker_api.rs:279] app 9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH run install cmd npm i --production --force, exit code 125
[2023-04-15 09:21:10.523383 +00:00] ERROR [ThreadId(8)] [service/app-manager/src/app_controller.rs:183] docker install failed. app:9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH failed, err: (Failed, app 9tGpLNniojkX6rVCdDi7t6BK1sMqDjw32RFhxQUReiFH run install cmd npm i --production --force, exit code 125, None)
```
[app-manager_3874026_rCURRENT.log](https://github.com/buckyos/CYFS/files/11238994/app-manager_3874026_rCURRENT.log)

**Comment by streetycat** [comment]
I meet the same error. and I have tried to run the command ` npm i --production --force` on the `OOD`, it's successed.

**Comment by weiqiushi** [comment]
it is the same error cause by #203

As a temporary solution, you can restart OOD and then reinstall the corresponding DecApp
@streetycat @lizhihongTest

**Comment by weiqiushi** [comment]
> > it is the same error cause by #203
> > As a temporary solution, you can restart OOD and then reinstall the corresponding DecApp @streetycat @lizhihongTest
> 
> No, It will not recover after the `OOD` restarted. and it's different with #203 .
> 
> there is no process for docker running.
> 
> ```
> root@mock-oodbox:~# sudo docker ps
> CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
> ```
Needing further information to analyze your problem, you need to upload the entire error app-manager log, here is the relevant docker command line

**Comment by weiqiushi** [comment]
> > > it is the same error cause by #203
> > > As a temporary solution, you can restart OOD and then reinstall the corresponding DecApp @streetycat @lizhihongTest
> > 
> > 
> > No, It will not recover after the `OOD` restarted. and it's different with #203 .
> > there is no process for docker running.
> > ```
> > root@mock-oodbox:~# sudo docker ps
> > CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES
> > ```
> 
> Needing further information to analyze your problem, you need to upload the entire error app-manager log, here is the relevant docker command line

I got the login for this OOD from another source and found that there is a problem with the docker environment on this OOD and I cannot pull the image needed for DecApp:

`failed to register layer: error creating overlay mount to /var/lib/docker/overlay2/ e669ae8312fc3e90f30916de047cc007deb54fd4b3b4f9788b3d01163f822836/merged: too many levels of symbolic links`

This issue is not related to AppManager

**Comment by weiqiushi** [comment]
It already fixed, see #217

**Comment by lizhihongTest** [comment]
DEC APP install command has support "npm i --production --force"

### Testing Environment

Nighlty ： 1.1.0.753-preview

OOD System : Ubuntu22

dec-app-install-timeout : cyfs://5r4MYfFJ7ktzBzSi1sWmU7BJLgpEEdp8ukbWemsFuQc1/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci

set install command:
```
"install": ["npm i --production --force", "node service/service.js --install"],
```
### Verification Step:

+ Install dec app cyfs://5r4MYfFJ7ktzBzSi1sWmU7BJLgpEEdp8ukbWemsFuQc1/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci
+ Check install result: Install DEC APP sucesss
+ Check install log:
```
[2023-04-23 12:07:50.247422 +08:00] INFO [ThreadId(11)] [service/app-manager/src/dapp.rs:110] load dapp package.cfg success, json content: Object {"executable": Array [], "id": String("dec-app-service"), "install": Array [String("npm i --production --force"), String("node service/service.js --install")], "start": String("node service/service.js --start"), "status": String("node service/service.js --status"), "stop": String("node service/service.js --stop"), "version": String("1.0.0")}
[2023-04-23 12:07:52.380739 +08:00] INFO [ThreadId(9)] [service/app-manager/src/dapp.rs:110] load dapp package.cfg success, json content: Object {"executable": Array [], "id": String("dec-app-service"), "install": Array [String("npm i --production --force"), String("node service/service.js --install")], "start": String("node service/service.js --start"), "status": String("node service/service.js --status"), "stop": String("node service/service.js --stop"), "version": String("1.0.0")}
[2023-04-23 12:07:52.452760 +08:00] INFO [ThreadId(9)] [service/app-manager/src/docker_api.rs:276] start app 9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci install cmd npm i --production --force in docker
[2023-04-23 12:07:52.521602 +08:00] INFO [ThreadId(9)] [service/app-manager/src/docker_api.rs:79] will spawn cmd: docker run --name decapp-9tgplnnhv6hh86h6y5slynbq6q4stttrgnslwtsyjtci-install --cap-add NET_ADMIN --cap-add NET_RAW --rm --init --network bridge -v /cyfs/log/app/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci:/cyfs/log/app -v /cyfs/data/app/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci:/cyfs/data/app/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci -v /cyfs/app/dockerfile/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci/start.sh:/opt/start.sh:ro -v /cyfs/tmp:/cyfs/tmp -v /etc/localtime:/etc/localtime:ro -v /etc/resolv.conf:/etc/resolv.conf:ro -v /cyfs/app/9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci:/opt/app buckyos/dec-app-base:v1 npm i --production --force
[2023-04-23 12:11:19.297240 +08:00] INFO [ThreadId(9)] [service/app-manager/src/docker_api.rs:312] app 9tGpLNnhV6Hh86h6y5SLYnBQ6Q4SttTrgnSLWTsyjTci run install cmd npm i --production --force success
[2023-04-23 12:11:24.227342 +08:00] INFO [ThreadId(8)] [service/app-manager/src/dapp.rs:110] load dapp package.cfg success, json content: Object {"executable": Array [], "id": String("dec-app-service"), "install": Array [String("npm i --production --force"), String("node service/service.js --install")], "start": String("node service/service.js --start"), "status": String("node service/service.js --status"), "stop": String("node service/service.js --stop"), "version": String("1.0.0")}
```
